core/src/accounts_hash_verifier.rs

```diff
@@ -161,7 +163,8 @@ impl AccountsHashVerifier {
                 == 0
             {
                 if let Some(pending_snapshot_package) = pending_snapshot_package {
-                    *pending_snapshot_package.lock().unwrap() = Some(accounts_package);
+                    *pending_snapshot_package.lock().unwrap() =
+                        Some(SnapshotPackage::FullSnapshotPackage(accounts_package));
```

Comment:
FYI: The larger `if` block will end up with an additional `else if` to see if AccountsHashVerifier should take in incremental snapshot here instead.